pickLayer/core/set_active_layer_tool.py

```diff
@@ -40,39 +40,42 @@ class SetActiveLayerTool(QgsMapToolIdentify):
     Map tool for selecting feature and its versions.
     """
 
-    def __init__(self, canvas: QgsMapCanvas, layer_type: str = "AllLayers") -> None:
+    def __init__(
+        self,
+        canvas: QgsMapCanvas,
+        layer_type: QgsMapToolIdentify.Type = QgsMapToolIdentify.AllLayers,
+    ) -> None:
 
-        self.layer_type = getattr(QgsMapToolIdentify, layer_type)
-        self.canvas = canvas
-        QgsMapToolIdentify.__init__(self, canvas)
+        self.layer_type = layer_type
+        super().__init__(canvas)
         self.setCursor(QCursor())
 
-    def canvasReleaseEvent(self, mouse_event) -> None:  # noqa N802
-        orig_search_radius = Settings.identify_tool_search_radius.get()
+    def searchRadiusMM(self) -> float:  # noqa N802
+        return Settings.search_radius.get()
+
+    def canvasReleaseEvent(self, mouse_event: QgsMapMouseEvent) -> None:  # noqa N802
         try:
-            search_radius = Settings.search_radius.get()
-            LOGGER.debug(f"Setting search radius to {search_radius}")
-            Settings.identify_tool_search_radius.set(search_radius)
 
             self.set_active_layer_using_closest_feature(
-                mouse_event.x(), mouse_event.y()
+                self._from_canvas_to_map_coordinates(mouse_event.x(), mouse_event.y())
```

Comment:
there is a method `self.toMapCoordinates(mouse_event.pos())` that can be used here, a custom method is not necessary

Reply:
fixed